app/src/app/components/sidebar/PopulationPanel/PopulationChart/PopulationLabels.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import {formatNumber} from '@/app/utils/numbers';
+import {Group} from '@visx/group';
+
+export const PopulationLabels: React.FC<{
+  xScale: (value: number) => number;
+  yScale: (value: number) => number;
+  entry: {zone: number; total_pop: number};
+  maxPop: number;
+  index: number;
+  barHeight: number;
+  isHovered: boolean;
+  showPopNumbers: boolean;
+  showTopBottomDeviation: boolean;
+}> = ({
+  xScale,
+  yScale,
+  entry,
+  maxPop,
+  index,
+  barHeight,
+  isHovered,
+  showPopNumbers,
+  showTopBottomDeviation,
+}) => {
+  const popDiffLabel = formatNumber(entry.total_pop - maxPop, 'string');
+  const popLabel = formatNumber(entry.total_pop, 'string');
+  return (
+    <Group
+      left={xScale(entry.total_pop)}
+      top={yScale(index) + barHeight}
+      style={{pointerEvents: 'none'}}
+    >
+      {!!(isHovered || showPopNumbers) && (
+        <text x={5} y={-2} fontSize={14} fontWeight={'bold'} textAnchor="start">
+          {popLabel}
+        </text>
+      )}
+      {!!(isHovered || showTopBottomDeviation) && (
```

Comment:
these look great in general- I did notice that for bars with small pops, the labels often fall off the side of the bar. What about checking the label width and not showing if it's below a certain threshold? 
```
const barX = xScale(entry.total_pop);
const labelWidth = 50; // or whatever; this worked in a quick test
const isPopDiffLabelVisible = barX - labelWidth > 0;
```
and then adding `isPopDiffLabelVisible` as a condition to render?

Reply:
This is a great catch, thank you! I added a check as you suggested in the [labels components](https://github.com/districtr/districtr-v2/blob/e72508803d0bf5e68b1eab799e9e559c61766557/app/src/app/components/sidebar/PopulationPanel/PopulationChart/PopulationLabels.tsx#L33-L38) that nudges the label group right if the bar is too short, and left in the probably unusual case that it is too long. 

<img width="497" alt="Screenshot 2024-12-17 at 9 34 26 AM" src="https://github.com/user-attachments/assets/48d7998f-ee79-41b5-9f5b-2a12068eb7ee" />

We _may_ need additional descriptive text to explain what these numbers mean, but my instinct is to keep it clean and get some user feedback